crates/project-model/src/cargo_workspace.rs

```diff
@@ -288,7 +288,24 @@ impl CargoWorkspace {
         locked: bool,
         progress: &dyn Fn(String),
     ) -> anyhow::Result<(cargo_metadata::Metadata, Option<anyhow::Error>)> {
-        Self::fetch_metadata_(cargo_toml, current_dir, config, sysroot, locked, false, progress)
+        let res = Self::fetch_metadata_(
+            cargo_toml,
+            current_dir,
+            config,
+            sysroot,
+            locked,
+            false,
+            progress,
+        );
+        if let Ok((_, Some(ref e))) = res {
+            tracing::warn!(
+                %cargo_toml,
+                %e,
+                "`cargo metadata` failed, but retry with `--no-deps` succeeded"
+            );
+            tracing::debug!("{e:?}");
```

Comment:
@Veykril I left the `tracing::debug!("{e:?}");` line in for now. Do you have an idea how to achieve detailed logging in debug mode without having two `tracing::` lines?

Reply:
I commented above, but see `tracing`'s documentation on `?` and `%` sigils: https://docs.rs/tracing/latest/tracing/#recording-fields. `tracing` doesn't have per-field levels, so you'll need to have two events or accept the higher verbosity.